Error code changed BAD_REQUEST to FORBIDDEN

b-thebest opened the pull request:
…tion_id/task/<task_id>/comments

### Description

<!--- Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change. --->

Fixes #636

### Type of Change:

<!--- **Delete irrelevant options.** --->

- Code: error code changed from BAD_REQUEST to FORBIDDEN

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)



### How Has This Been Tested?
Tested locally 

<!-- Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test. -->


### Checklist:

<!-- **Delete irrelevant options.** -->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

b-thebest committed:
fix: Error code is changed
In task_comment.py error code changed from BAD_REQUEST to FORBIDDEN for mentorship_relation/relation_id/task/<task_id>/comments

devkapilbansal reviewed (dismissed):
LGTM :+1:

devkapilbansal commented:
@b-thebest please update your PR title and commit message by following the Commit message guidelines

b-thebest commented:
@devkapilbansal I have done some changes, These changes are correct? Please correct me if I am wrong

devkapilbansal reviewed (approved):
LGTM :+1: 
Thanks @b-thebest  for this

vj-codes reviewed (dismissed):
@b-thebest thank you for your contribution:)
LGTM 🎉

devkapilbansal commented:
My bad, @b-thebest this PR needs changes

devkapilbansal reviewed (changes requested):
Change this to Forbidden too
https://github.com/anitab-org/mentorship-backend/blob/fffb4c39027b042f392d4edf0bb462e522cb7320/app/api/dao/task_comment.py#L21

b-thebest committed:
fix: Error code is changed in dao
In dao/task_comment.py error code return status changed from BAD_REQUEST to FORBIDDEN for mentorship_relation/relation_id/task/<task_id>/comments

b-thebest commented:
@devkapilbansal It's done, but some checks are getting failed. Is it fine ?

devkapilbansal reviewed (dismissed):
LGTM 👍 
@b-thebest codecov/patch failed because these lines are not covered. #989 in already opened for that

devkapilbansal commented:
@vj-codes please review it again

epicadk reviewed (dismissed):
💯

devkapilbansal commented:
@b-thebest one last request.
Please rebase your branch with current develop. Then you may need to make changes in some tests too.

devkapilbansal commented:
Once that is done, I will add label `Ready to Merge` since I tested too and automated tests will also ensure its working

b-thebest commented:
@devkapilbansal Sorry for being late. I will do it and let you know soon

b-thebest commented:
> @b-thebest one last request.
> Please rebase your branch with current develop. Then you may need to make changes in some tests too.

Could you please help me in that, sometimes I get stuck in rebasing.

vj-codes commented:
@b-thebest any updates here?

b-thebest commented:
I have made pull request. Could you please help me in rebasing my
repository with remote ?

On Wed, May 5, 2021, 11:07 PM Vaishnavi Joshi ***@***.***>
wrote:

> @b-thebest <https://github.com/b-thebest> any updates here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/anitab-org/mentorship-backend/pull/1030#issuecomment-832879835>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ6VVHY2Y3NWWQ2MZAFWDOTTMF65ZANCNFSM4YSXRAEA>
> .
>

vj-codes commented:
@b-thebest `git checkout develop` and then `git merge branchname `should work ?
cc : @epicadk

b-thebest commented:
My changes in new branch and pull request will still be there after
rebasing?

On Wed, May 5, 2021, 11:22 PM Vaishnavi Joshi ***@***.***>
wrote:

> @b-thebest <https://github.com/b-thebest> git checkout develop and then git
> merge branchname should work ?
> cc : @epicadk <https://github.com/epicadk>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/anitab-org/mentorship-backend/pull/1030#issuecomment-832889509>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ6VVH76WFGACQGZ76PBNHLTMGAWZANCNFSM4YSXRAEA>
> .
>

devkapilbansal commented:
@b-thebest to rebase your branch just follow:-
```
git fetch upstream develop
git rebase upstream/develop
git push -f
```

Your PR and changes done will remain after rebasing

vj-codes commented:
@b-thebest any updates? you can also do this via github

b-thebest commented:
Sorry for being late. My health conditions were not good during past days. I will be completing it very soon and share update as soon as possible.

b-thebest committed:
fix: status code has been changes from BAD_REQUEST to FORBIDDEN

b-thebest commented:
Please review this PR and close the issue

vj-codes reviewed (dismissed):
Thank you for your contribution @b-thebest 🎉

devkapilbansal commented on app/api/resources/task.py:
```diff
@@ -23,7 +23,7 @@ class CreateTask(Resource):
     @task_ns.expect(auth_header_parser, create_task_request_body)
     @task_ns.response(HTTPStatus.CREATED, f"{messages.TASK_WAS_CREATED_SUCCESSFULLY}")
-    @task_ns.response(HTTPStatus.FORBIDDEN, f"{messages.UNACCEPTED_STATE_RELATION}")
+    @task_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.UNACCEPTED_STATE_RELATION}")
```
Why this change is made??

b-thebest replied:
I think unaccepted relation must fall in BAD_REQUEST. When unauthorized user try to do something then FORBIDDEN is required

epicadk replied:
Nope should be forbidden. Nice catch @devkapilbansal . Bad request would imply that the server cannot understand the request, however here the server understands the request but refuses to process it.

b-thebest replied:
I changed it again to FORBIDDEN. Please check
@devkapilbansal @epicadk

b-thebest committed:
Change from BAD_REQUEST to FORBIDDEN in task.py

devkapilbansal reviewed (dismissed):
LGTM :+1: 

Thanks @b-thebest for your contribution

isabelcosta reviewed (dismissed):
Great work @b-thebest ! Thank you for contributing!

isabelcosta committed:
Merge branch 'develop' into task-comment-error-code

isabelcosta commented:
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses (positive and negative tests) were tested as below:

   * _Get comments from a task of a non accepted relation_  
     _Screenshot/gif_:  
![image](https://user-images.githubusercontent.com/11148726/126714858-2b53fb9c-daec-45d4-a8dc-df8eeff91c37.png)

     _Expected Result_:  403 is returned when fetching comments from a not accepted relation tasks
     _Actual Result_: 403 is returned as expected

  
4. Additional Comments: https://ms-backend-review-pr-1030.herokuapp.com/

5. Status of PR Changed to: Ready to Merge.

isabelcosta commented:
will merge, because the approvals were revoked by my merge conflict resolve
